.github/workflows/ci.yml

```diff
@@ -13,159 +13,103 @@ concurrency:
-
-    - name: Set up checkout
-      run: |
-        cp ./scripts/standalone.gclient .gclient
-
-    - name: gclient sync --no-history --shallow
-      run: gclient sync --no-history --shallow
-      env:
-        DEPOT_TOOLS_WIN_TOOLCHAIN: 0
-
-    - name: Set up sccache
-      uses: mozilla-actions/sccache-action@v0.0.3
-
-    - name: Generate build files
-      shell: bash
-      run: |
-        mkdir -p ${{ steps.strings.outputs.build-output-dir }}
-        cat << EOF >> ${{ steps.strings.outputs.build-output-dir }}/args.gn
-        cc_wrapper="sccache"
-        is_debug=false
-        is_component_build=true
-        is_clang=true
-        EOF
-        gn gen ${{ steps.strings.outputs.build-output-dir }}
-
-      env:
-        SCCACHE_GHA_ENABLED: "true"
-        DEPOT_TOOLS_WIN_TOOLCHAIN: 0
-
-    - name: Build
-      run: autoninja -C ${{ steps.strings.outputs.build-output-dir }}
-      env:
-        SCCACHE_GHA_ENABLED: "true"
-        DEPOT_TOOLS_WIN_TOOLCHAIN: 0
-
   cmake:
     strategy:
       # Set fail-fast to false to ensure that feedback is delivered for all matrix combinations. Consider changing this to true when your workflow is stable.
       fail-fast: false
 
+      # Set up a matrix to run the following 6 configurations:
+      # 1. <Windows, Debug, latest MSVC compiler toolchain on the default runner image, default generator>
+      # 2. <Windows, Release, latest MSVC compiler toolchain on the default runner image, default generator>
+      # 3. <Linux, Debug, latest GCC compiler toolchain in the container, default generator>
+      # 4. <Linux, Release, latest GCC compiler toolchain in the container, default generator>
+      # 5. <MacOS, Debug, latest Clang compiler toolchain on the default runner image, default generator>
+      # 6. <MacOS, Release, latest Clang compiler toolchain on the default runner image, default generator>
       matrix:
-        os: [ubuntu-latest, windows-latest, macos-latest]
+        os: [macos-14, ubuntu-22.04, windows-2022]
         build_type: [Debug, Release]
-        c_compiler: [clang, cl]
+        toolchain: [gcc, clang, msvc]
         include:
-          - os: windows-latest
+          - os: macos-14
+            toolchain: clang
+            c_compiler: $(brew --prefix llvm@15)/bin/clang
+            cpp_compiler: $(brew --prefix llvm@15)/bin/clang++
+            env:
+              MACOSX_DEPLOYMENT_TARGET: "11.0"
+            container: null
+          - os: windows-2022
+            toolchain: msvc
             c_compiler: cl
             cpp_compiler: cl
-          - os: ubuntu-latest
-            c_compiler: clang
-            cpp_compiler: clang++
-          - os: macos-latest
-            c_compiler: clang
-            cpp_compiler: clang++
+            container: null
+          - os: ubuntu-22.04
+            toolchain: gcc
+            c_compiler: gcc
+            cpp_compiler: g++
+            container: dockcross/manylinux_2_28-x64:latest
```

Comment:
Any reason to use that container instead of running on some Ubuntu?

Reply:
Yes, for ABI compatibility with glibc 2.28. This way, the cd mechanism casts a wide net across many linux distros. IIUC, it is the same reason wgpu-native also compiles their implementation in manylinux_2_28. For reference - https://github.com/gfx-rs/wgpu-native/blob/trunk/.github/workflows/cd.yml#L5

Reply:
Ok, can you add a comment for this? It would help folks changing the actions in the future understand why we do things.

Reply:
Done